system/system_windows.go

```diff
@@ -72,67 +73,120 @@ func removeUninstallEntry(name string) error {
+			displayName, _, err := q.GetStringValue("DisplayName")
 			if err != nil {
 				continue
 			}
-			// Remove display name whitespace to conform to package name
-			if strings.ToLower(publisherReg) == strings.ToLower(publisher) {
-				displayName, _, _ := q.GetStringValue("DisplayName")
-				if strings.Contains(strings.ToLower(programName), strings.ToLower(strings.ReplaceAll(displayName, " ", ""))) {
-					// Check if the value exists, move on if it doesn't
-					un, _, err := q.GetStringValue("QuietUninstallString")
+
+			switch extension {
+			case ".msi":
+				a, _, err := q.GetStringValue("InstallSource")
+				if err != nil {
+					// InstallSource not found, move on to next entry
+					continue
+				}
+				iS := strings.Split(installSource, "@")
+				if strings.Contains(a, iS[0]) {
+					name, _, err := q.GetStringValue("DisplayName")
 					if err != nil {
-						un, _, err = q.GetStringValue("UninstallString")
-						if err != nil {
-							// UninstallString not found, move on to next entry
-							continue
-						}
+						// UninstallString not found, move on to next entry
+						continue
 					}
-					return un
+					return name, productReg
+				}
+			default:
+				for _, v := range publisherNameReg {
+					re := regexp.MustCompile("(?i)" + regex[v])
+					publisher = re.ReplaceAllString(publisher, "")
+				}
+				for _, v := range programNameReg {
+					re := regexp.MustCompile("(?i)" + regex[v])
+					programName = re.ReplaceAllString(programName, "")
+				}
+				// Ignore empty and googet labeled pacakges
+				if displayName == "" || strings.Contains(displayName, "GooGet -") {
+					continue
+				}
+				// Check if Package name is in display name removing spaces
+				if strings.Contains(strings.ToLower(strings.ReplaceAll(displayName, " ", "")), strings.ToLower(programName)) {
+					// Check if the value exists, move on if it doesn't
+
+					return displayName, productReg
+				}
+				// Check if Package name is in display name removing dashes
+				if strings.Contains(strings.ToLower(strings.ReplaceAll(displayName, "-", "")), strings.ToLower(programName)) {
+					// Check if the value exists, move on if it doesn't
+					return displayName, productReg
+				}
+				if strings.Contains(strings.ToLower(programName), strings.ToLower(strings.ReplaceAll(displayName, " ", ""))) {
+					// Check if the value exists, move on if it doesn't
+					return displayName, productReg
+				}
+				a, _, err := q.GetStringValue("InstallSource")
+				if err != nil {
+					// InstallSource not found, move on to next entry
+					continue
+				}
+				iS := strings.Split(installSource, "@")
+				if strings.Contains(a, iS[0]) && installSource != "" {
+					fmt.Println(installSource)
+					return displayName, productReg
 				}
 			}
+
 		}
+
+	}
+	return "", ""
+}
+
+func uninstallString(regkey, extension string) string {
+	q, err := registry.OpenKey(registry.LOCAL_MACHINE, regkey, registry.ALL_ACCESS)
+	defer q.Close()
+	if err != nil {
+		fmt.Printf("%v", err)
```

Comment:
This should be a log if you don't care about it, or return the error if you do; same for the other printf below.